### DISCUSSION TITLE
Call an API within getServerSideProps or just access the database

### DISCUSSION THREAD

**Discussion by onefineplayandy** [vercel/next.js#15750] [Help] [answered]
I'm in a bit of a quandry at the moment and i wonder if someone can help me figure this out. 

I have a simple application. You login, then it should take you to a 'userdetails/create' page allowing the user to populate some more data. 

My current plan is to utilise getServerSideProps to:
- ensure user is logged in 
- get existing user details
- return to page as props

At the moment i'm using an API to get the existing user details. But all this does is: 
- ensure user is logged in 
- access the database do a select
- return some data 
- or various error types

I guess my question is. What is the point of doing the API call if all it's doing is accessing a database on the serverside. Why dont i just access the database directory to solve my problem? This would save me
- doubling up the authentication check 
- reduce network costs as i wont need to do an api call 

The page, api and also the DB connection / user details service layer are all within the same codebase.

**Comment by jamesmosier** [comment, marked as answer]
Sounds like you have your answer 😄  if all of your application logic is self contained already in your project, I'd say just include your logic in `getServerSideProps` if it is only relevant for that page it is used in and will only be consumed internally within your application. 

I don't think you can go wrong, but it sounds like you have more pros than cons for putting it in `getServerSideProps`.

Just my 2 cents 🚀

**Reply by onefineplayandy** [reply]
i think you're right, it makes sense to in serverSideProps to just get the data nad on the client side to use an API if i need to (forms or whatever) thanks for helping me clarify this in my head.
